### DISCUSSION TITLE
Index display disturbed due to conditional "Badge Field"

### DISCUSSION THREAD

**Discussion by mahfoudfx** [laravel/nova-issues#3552] [Q&A]
- Laravel Version: 8.59.0
- Nova Version: 3.29.0
- PHP Version: 8.0.10
- Database Driver & Version:
- Operating System and Version: Ubuntu 20.04.3 LTS
- Browser type and version: Microsoft Edge 93.0.961.38 / Google Chrome 93.0.4577.63

### Description:
Hello,
I have placed a `Badge Field` for payments status with condition to show on `Index` only when it is not paid. I get the below index display.
Figure 1: when first row meets the condition to display the `badge field` on `index`.
![image](https://user-images.githubusercontent.com/31103481/133251301-7f678161-c553-425c-97b9-55a195af46fb.png)

Figure 2: when the first row doesn't meet the condition to display the `badge field` on `index`.
![image](https://user-images.githubusercontent.com/31103481/133251562-90ce09bc-e8e8-48ef-be52-ec6552f41346.png)

### Detailed steps to reproduce the issue on a fresh Nova installation:
Place a `Badge Field` with condition to display on `Index`.

**Comment by crynobone** [comment]
Please provide full repository to help us debug your exact use case.

**Comment by mahfoudfx** [comment]
Below the part of code in the `Resource`

```php
(new Panel('Payment', $this->paymentFields()))->limit(2),
//
//Some code
//
protected function paymentFields()
    if ($this->price_tf - $this->paid_amount >0)
        return [
            Badge::make('Payment', function () {
                return 'Not paid';
                })->map([
                    'Not paid' => 'danger',
                ]),
        ];
        else
        return [
            Badge::make('Status', function () {
                return 'Paid';
                })->map([
                    'Paid' => 'success',
                ])->hideFromIndex(),
}
```
